common/dependabot-common.gemspec

```diff
@@ -20,8 +20,8 @@ Gem::Specification.new do |spec|
   }
 
   spec.version = Dependabot::VERSION
-  spec.required_ruby_version = ">= 3.1.0"
-  spec.required_rubygems_version = ">= 3.3.7"
+  spec.required_ruby_version = ">= 3.2.0"
```

Comment:
I realize this is a little opinionated given this is a library, but still, I think it's not too unreasonable... `3.2` has been out for a while and wasn't that large of a jump from `3.1`, so shouldn't be that hard for folks to upgrade their toolchains unless they're airgapped/really monolithic environments with build systems that are super coupled... but I generally doubt folks run Dependabot tightly coupled, since they can run Dependabot as a sidecar script using it's own mini environment.